fixed database

johntheagent2 opened the pull request.

johntheagent2 committed:
fixed database

ngahpsparkminds commented on src/main/java/org/example/librarymanagement/service/implement/AppUserServiceImpl.java:
```diff
@@ -46,19 +48,27 @@ public void resetWrongLoginCounter(String email) {
 
+    @Transactional
     @Override
     public void resetPassword(String token) {
```
Chỗ này chắc em truyền thêm email để query tránh trường hợp token generate bị duplicate

johntheagent2 committed:
get email from context instead of letting user do input,
added verify by user email in tokenOtpService,
added jwt token expired exception handler
